src/EFCore.PG/Storage/Internal/NpgsqlDatabaseCreator.cs

```diff
@@ -178,6 +178,15 @@ await unpooledRelationalConnection.OpenAsync(errorsExpected: true, cancellationT
 
             throw;
         }
+        catch (NpgsqlException e) when (
+            // This can happen when Npgsql attempts to connect to multiple hosts
+            e.InnerException is AggregateException ae &&
+            ae.InnerExceptions.Any(
+                ie => ie is PostgresException pe
+                && IsDoesNotExist(pe)))
```

Comment:
tiny non-mandatory nit suggestion:

```suggestion
            ae.InnerExceptions.Any(ie => ie is PostgresException pe && IsDoesNotExist(pe)))
```